lib/ts/ink_mutex.h

```diff
@@ -48,61 +49,57 @@ class x_pthread_mutexattr_t
 {
 public:
   pthread_mutexattr_t attr;
+
   x_pthread_mutexattr_t();
-  ~x_pthread_mutexattr_t() {}
+  ~x_pthread_mutexattr_t();
 };
-inline x_pthread_mutexattr_t::x_pthread_mutexattr_t()
-{
-  pthread_mutexattr_init(&attr);
-#ifndef POSIX_THREAD_10031c
-  pthread_mutexattr_setpshared(&attr, PTHREAD_PROCESS_SHARED);
-#endif
-}
 
-extern class x_pthread_mutexattr_t _g_mattr;
-
-static inline int
-ink_mutex_init(ink_mutex *m, const char *name)
+static inline void
+ink_mutex_init(ink_mutex *m, const char * /* name */)
 {
-  (void)name;
+  int error;
+  x_pthread_mutexattr_t attr;
```

Comment:
Hmmm. I don't see the benefit of constructing `attr` for every mutex initialization, especially since ATS creates a lot of mutexes. I'd leave it locally scoped and just mark it static.